medusa/reconstruct/expand.py

```diff
@@ -1,261 +1,72 @@
-    # error.
-    exchanges_in_phenotype_dict = set()
-    for condition in phenotype_dict.keys():
-        exchanges_in_phenotype_dict = exchanges_in_phenotype_dict | set(
-                                        phenotype_dict[condition].keys())
-
-    model_rxns = [rxn.id for rxn in model.reactions]
-    for exchange in exchanges_in_phenotype_dict:
-        if not exchange in model_rxns:
-            print('Could not find '+ exchange)
-            raise ValueError('phenotype_dict contains exchange reactions not '
-                            'found in model. Add any necessary exchange '
-                            'reactions prior to gapfilling.')
-
-
-    # pre-generate the random orderings of conditions to ensure there are no
-    # duplicates.
-    cycle_order = []
-    while len(cycle_order) < output_ensemble_size:
-        condition_list = random.sample(list(phenotype_dict.keys()),
-                                        len(phenotype_dict.keys()))
-        if condition_list not in cycle_order:
-            cycle_order.append(condition_list)
-
-    # Cycle through all conditions and gapfill. After each gapfill iteration,
-    # our strategy is to reduce the cost for the reactions returned by the
-    # previous solution to 0, such that they are automatically included in
-    # the model for the next condition.
-    if gapfill_type is "integer":
-        solutions =  _integer_iterative_binary_gapfill(model,
-                              phenotype_dict,
-                              cycle_order,
-                              universal=universal,
-                              output_ensemble_size=output_ensemble_size,
-                              lower_bound=lower_bound,
-                              penalties=penalties,
-                              demand_reactions=demand_reactions,
-                              exchange_reactions=exchange_reactions,
-                              integer_threshold=inclusion_threshold)
-    elif gapfill_type is "continuous":
-        solutions = _continuous_iterative_binary_gapfill(model,
-                              phenotype_dict,
-                              cycle_order,
-                              universal=universal,
-                              output_ensemble_size=output_ensemble_size,
-                              lower_bound=lower_bound,
-                              penalties=penalties,
-                              demand_reactions=demand_reactions,
-                              exchange_reactions=exchange_reactions,
-                              flux_cutoff=inclusion_threshold,
-                              exchange_prefix='EX_')
-
-    ensemble =_build_ensemble_from_gapfill_solutions(model,solutions,
-                                                    universal=universal)
-    return ensemble
-
-def _continuous_iterative_binary_gapfill(model,phenotype_dict,cycle_order,
+def _continuous_iterative_binary_gapfill(model = None, 
```

Comment:
The probabilistic gap filling functionality should be added to the gapfill_to_ensemble function instead of _continuous_iterative_binary_gapfill (which is for gapfilling when multiple media conditions are available for gap filling, such as in Biolog growth assays).